flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adapter/DefaultResultPartition.java

```diff
@@ -42,17 +43,21 @@
 
 	private final ResultPartitionType partitionType;
 
+	private final Supplier<ResultPartitionState> resultPartitionStateSupplier;
+
 	private DefaultExecutionVertex producer;
 
 	private final List<DefaultExecutionVertex> consumers;
 
 	DefaultResultPartition(
 			IntermediateResultPartitionID partitionId,
 			IntermediateDataSetID intermediateDataSetId,
-			ResultPartitionType partitionType) {
+			ResultPartitionType partitionType,
+			Supplier<ResultPartitionState> resultPartitionStateSupplier) {
```

Comment:
Just an observation: Isn't this class now very similar to `TestingSchedulingResultPartition`?

Reply:
Yes it's very similar. 
Actually I'm thinking of simplifying the entire TestingSchedulingTopology which is redundant since it is used to mock the jobVertex->executionVertex structure of ExecutionGraph. 
I think we can make it much simpler like `TestFailoverTopology`, which would further make  related tests simpler.